iocore/net/SSLSessionCache.h

```diff
@@ -198,3 +201,19 @@ class SSLSessionCache
   SSLSessionBucket *session_bucket = nullptr;
   size_t nbuckets;
 };
+
+class SSLOriginSessionCache
+{
+public:
+  SSLOriginSessionCache();
+  ~SSLOriginSessionCache();
+
+  void insertSession(std::string lookup_key, SSL_SESSION *sess);
+  SSL_SESSION *getSession(std::string lookup_key);
+
+private:
+  mutable std::shared_mutex mutex;
```

Comment:
Would you share me the background of using `std::shared_mutex` instead of `pthread_mutex` a.k.a. `ink_mutex`?

Reply:
Well, since the server side session cache has proven that using `std::shared_mutex` is beneficial, I thought it should be used here as well.

Reply:
For our architectures, pthread_mutex and std::shared_mutex should be equivalent, but moving forward std::shared_mutex should give us better portability and ability to pick up new mutex technology as it becomes available.